### PR TITLE
Added Phone Number Patterns on install for US, UK, FR and EN 

### PR DESCRIPTION
**Pull request by Undefined-Variables**
Added phone number input types for United States, United Kingdom, France and Spain, My code is to the correct standard and I have tested each option. It happens during install and I have added radio buttons to the install script, the style needs looking at as rules already in place are not possible to avoid without messing up preexisting form elements. Could have used check boxes but in order to only allow a single choice would have taken a php script in the middle of install form so odiously wanted to avoid that. 

The script is fully working and regex standard expressions are what control the input. I have included french and Spanish language but its been a while since I wrote in either nut tried :)

Looking forward to hearing your thoughts on the changes.

### CODE REVIEW

File: InstallController.php [hunk 1]
```diff
@@ -0,0 +1,785 @@
...
+        }
+
+        $this->oView->assign('sept_number', 3);
+        $this->oView->assign('errors', @$aErrors);
+        unset($aErrors);
+        $this->oView->display('config_path.tpl');
+    }
+
+    /********************* STEP 4 *********************/
+    public function config_system()
```
[the remark this answers is not in the document]
Comment (Undefined-Variables) [reply InstallController.php]: That was not edited by me.

**Comment by polynamaude** [comment]
https://ph7cms.com/doc/en/code-convention

**Comment by Undefined-Variables** [comment]
I didn't edit the misses parts that was already like that and the edits I made followed the code practices, any way I am getting better and almost got that one perfect! NitPic found two genuine mistakes, a space missing after an if and it found 24 lines instead of 22! I think that was pretty good to be fair! :~p

Let me just laugh you said remember? I will get there and I am not a million miles away!

**Comment by Undefined-Variables** [comment]
You can reject this if its not up to standard you know. I won't be offended, only it's been here for days!

File: InstallController.php [hunk 2]
```diff
@@ -0,0 +1,785 @@
...
+                $this->oView->assign('failure', 1);
+            }
+        }
+
+        $this->oView->assign('sept_number', 2);
+        $this->oView->display('license.tpl');
+    }
+
+    /********************* STEP 3 *********************/
+    public function config_path()
```
[the remark this answers is not in the document]
Comment (rondlite) [reply InstallController.php]: whoever wrote the CI ... should know it is called UpperCamelCase  but ok..

**Comment by Undefined-Variables** [comment]
@rondlite Just read your comment in my emails but cant see it here, wired! I honestly didn't think to check the existing work!  My mistake and from now on I will check everything! It was only a little 10minute play around for my first ever contribution to here, next time I will make sure its something worth it. I don't know why its not been cancelled cos there are mistakes in it. 

Just wanted to see how it would be received, lets just say I wont be rushing to add anything else! Although that being said I am currently writing a module for the web cam chat but I don't see the point in posting. You get no thanks and everyone tells you how stupid you are when you make a mistake! This way of learning is hard work! Possibility and encouragement are key to learning anything!

**Comment by polynamaude** [comment]
@Undefined-Variables We are not in a play park. Look at how you treat others and this will give you a hint why this happens to you.

When people don't know and want to learn, we are more than happy to help. When people start accusing people of things that don't fit together it just stop from being possible to work together. Do you really think we have plenty of free time just to look after some **_Pull Request_** because you want to try around what it's like ?

There's two type of mistake or errors in life. The first one are caused because of a lack of knowledge, so you learn and go forward.

The second ones are caused by our inner way of reacting. For example, going off-topic many times and continuing even when people do tell you. And this also include opening off topic issues, asking questions that go answered, not reading manuals, etc.

If you think I'm just a sad person complaining at you then I've copied some links over here. They are some templates for issues. I suggest you look at them to give you an idea I ain't the only one who think this way. Even if most people will just ignore you.

https://github.com/stevemao/github-issue-templates/blob/master/no-duplicates/ISSUE_TEMPLATE.md

https://github.com/stevemao/github-issue-templates/blob/master/bugs-only/ISSUE_TEMPLATE.md

https://github.com/stevemao/github-issue-templates/blob/master/checklist/ISSUE_TEMPLATE.md
https://github.com/stevemao/github-issue-templates/blob/master/checklist/PULL_REQUEST_TEMPLATE.md

Go to your local computer club and make friends there, that is the best thing you can think about.

**Comment by Undefined-Variables** [comment]
I have apologised for the past and have been nothing but respectful to you ever since! I am a strong believer in second chances and I know there is no such thing as perfection. I was wrong and I have my reasons for behaving in the way I did, I was clearly not thinking straight and when you spoke to me the way you did I lost my cool! I am not proud of my actions but I am genuine when I say I am not a bad person and want to make amends. 

I am willing to listen and learn from people who know more than myself and take everything you say very seriously! I told you I have a lot of respect for you and the knowledge you process is not something you can get from a local computer club. I do have my uses, I don't want to sound arrogant but there is not a lot I can not do. I am a qualified mechanic, plumber and have over 90 certificates in computer science. Most of these are Clat, Photoshop, Acrobe, Mysql etc. I am not stupid by a long shot and while I was inside (don't ask, SR) I attended a computer club and the truth of the matter is I was teaching the professor more than he was teaching me.

I want to become good at this and I am willing to put the time and effort in to doing it correctly. I read everything you point me at and have been reading books on JavaScript as you suggested. 

Truth is I look up to people like yourself and made a mistake by letting myself get wound up. I will never talk to you like that ever again you have my word. Can you please let me start a fresh with you. I will delete this account and start again!

**Comment by Undefined-Variables** [comment]
These links are all the same?

**Comment by polynamaude** [comment]
@Undefined-Variables You don't get it, do you ? You are repeating this over and over again, saying the same thing. You don't seem to understand what is written here.

Life doesn't work like this and not even if you try the old trick of hiding behind a computer screen with a new account.

Go to your local computer club and make friend, that's it. This is a support forum for pH7CMs. What doesn't you understand ?

And as I'm writing you confirm exactly what I said. By asking me if those links are the same without taking the time to look at them.... And afterward the list of sorry, sorry, sorry, sorry...

**Comment by Undefined-Variables** [comment]
I am working on the software and I told you before I don't hide behind keyboards! That is not my style!

You must have something nice in you! Nobody is this cold. I didn't just argue with myself you know... It takes two to tango and I played my part but I was not alone! Shit man take some responsibility!

Whats it like up there on your high horse hiding behind a  keyboard and multiple aliases?

File: InstallController.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (polynamaude) [reply InstallController.php]: Ha ha ha "camel caps", I think I'm going to be so surprised to day this dude will be able to use domain specific language in the proper context.

**Pull request closed without merging**